test/test-cxx.jl

```diff
@@ -41,3 +41,30 @@ run(`c++ -shared -o libAddIntegersCxx.$dlext AddIntegersCxx.o`)
 if Sys.ARCH ≡ :x86_64
     @test AddIntegersCxx.add_int(2, 3) == 5
 end
+
+@testset "cxxtype" begin
+    for T in [
+        Cstring,
```

Comment:
This is a complete list of the C interop types defined in julia.

Reply:
I don't think this test is very useful by itself; it essentially just repeats the definition of `cxxtype`. It would be more interesting to actually test code generation for all the types, maybe passing values of the respective type forth and back between Julia and C++.

For example, `add_int` could be generalized to `add` for all the arbitrary arithmetic types. String and pointer types would need to be tested differently.

If you do so, then I would keep the file `test-cxx.jl` as is since it also serves as a simple example, and create a new file `test-cxx-types.jl`.

Reply:
> I don't think this test is very useful by itself; it essentially just repeats the definition of cxxtype. 

At least for me, this is not obvious. Note that the `cxxtype` definition uses julia types like `Float32`, while this test uses C names like `float`. For instance, there is no `Int128` in the dict. For me it is still unknown if some of type in the test might alias to `Int128` in some exotic settings. Anyway I agree that a back and forth test is more useful. I will add one.